plugins/action/system/shutdown.py

```diff
@@ -0,0 +1,234 @@
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import random
+import time
+
+from datetime import datetime, timedelta
+
+from ansible.errors import AnsibleError, AnsibleConnectionFailure
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.common.collections import is_string
+from ansible.module_utils.common.validation import check_type_str
+from ansible.plugins.action import ActionBase
+from ansible.utils.display import Display
+
+display = Display()
+
+
+class TimedOutException(Exception):
+    pass
+
+
+class ActionModule(ActionBase):
+    TRANSFERS_FILES = False
+    _VALID_ARGS = frozenset((
+        'connect_timeout',
+        'msg',
+        'delay',
+        'search_paths'
+    ))
+
+    DEFAULT_CONNECT_TIMEOUT = None
+    DEFAULT_PRE_SHUTDOWN_DELAY = 0
+    DEFAULT_SHUTDOWN_MESSAGE = 'Shut down initiated by Ansible'
+    DEFAULT_SHUTDOWN_COMMAND = 'shutdown'
+    DEFAULT_SHUTDOWN_COMMAND_ARGS = '-h {delay_min} "{message}"'
+    DEFAULT_SUDOABLE = True
+
+    DEPRECATED_ARGS = {}
+
+    SHUTDOWN_COMMANDS = {
+        'alpine': 'poweroff',
+        'vmkernel': 'halt',
+    }
+
+    SHUTDOWN_COMMAND_ARGS = {
+        'alpine': '',
+        'void': '-h +{delay_min} "{message}"',
+        'freebsd': '-h +{delay_sec}s "{message}"',
+        'linux': DEFAULT_SHUTDOWN_COMMAND_ARGS,
+        'macosx': '-h +{delay_min} "{message}"',
+        'openbsd': '-h +{delay_min} "{message}"',
+        'solaris': '-y -g {delay_sec} -i 5 "{message}"',
+        'sunos': '-y -g {delay_sec} -i 5 "{message}"',
+        'vmkernel': '-d {delay_sec}',
+        'aix': '-Fh',
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(ActionModule, self).__init__(*args, **kwargs)
+
+    @property
+    def delay(self):
+        return self._check_delay('delay', self.DEFAULT_PRE_SHUTDOWN_DELAY)
+
+    def _check_delay(self, key, default):
+        """Ensure that the value is positive or zero"""
+        value = int(self._task.args.get(key, self._task.args.get(key + '_sec', default)))
+        if value < 0:
+            value = 0
+        return value
+
+    def _get_value_from_facts(self, variable_name, distribution, default_value):
+        """Get dist+version specific args first, then distribution, then family, lastly use default"""
+        attr = getattr(self, variable_name)
+        value = attr.get(
+            distribution['name'] + distribution['version'],
+            attr.get(
+                distribution['name'],
+                attr.get(
+                    distribution['family'],
+                    getattr(self, default_value))))
+        return value
+
+    def get_shutdown_command_args(self, distribution):
+        args = self._get_value_from_facts('SHUTDOWN_COMMAND_ARGS', distribution, 'DEFAULT_SHUTDOWN_COMMAND_ARGS')
+        # Convert seconds to minutes. If less that 60, set it to 0.
+        delay_min = self.delay // 60
```

Comment:
How about storing the value of `self.delay` as `delay_sec`, so you don't have to call `self.delay` a second time in this function?

Reply:
Fixed in https://github.com/ansible-collections/community.general/pull/700/commits/d275da6a06707613c0cef75e8e6cac0b7b7173a2.